### PR TITLE
refactor: simplify `CheckedAmountOf`

### PR DESCRIPTION
**Pull request by gregorydemay**
Follow-up on #275 to remove all methods on `CheckedAmountOf` that were not used. Since none of the checked operations are needed, the type was rename to `Amount`.

### CODE REVIEW

File: src/rpc_client/amount/mod.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
+use evm_rpc_types::Nat256;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::cmp::Ordering;
+use std::fmt;
+use std::marker::PhantomData;
+
+/// `CheckedAmountOf<Unit>` provides a type-safe way to keep an amount of some `Unit`.
+pub struct AmountOf<Unit>(ethnum::u256, PhantomData<Unit>);
+
+impl<Unit> AmountOf<Unit> {
```
Comment (rvanasa) [inline src/rpc_client/amount/mod.rs:10]: Subjective: maybe we could change this to `Amount<Unit = ()>`? That way, we could specify `Amount` in place of `AmountOf<()>`, and (imo) this also makes sense in contexts such as `type GasAmount = Amount<GasUnit>`.

**Review by rvanasa — APPROVED** [review]

File: src/rpc_client/amount/mod.rs (hunk 1 above)
Comment (gregorydemay) [reply src/rpc_client/amount/mod.rs]: Good idea, renamed.

**Comment by rvanasa** [comment]
Ran `cargo clippy --fix` and `cargo fmt` and pushed changes for CI.

File: src/rpc_client/amount/mod.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
+use evm_rpc_types::Nat256;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::cmp::Ordering;
+use std::fmt;
+use std::marker::PhantomData;
+
+/// `CheckedAmountOf<Unit>` provides a type-safe way to keep an amount of some `Unit`.
```
Comment (THLO) [inline src/rpc_client/amount/mod.rs:7]:
```suggestion
/// `Amount<Unit>` provides a type-safe way to keep an amount of some `Unit`.
```

**Review by THLO — APPROVED** [review]

**Comment by THLO** [comment]
Looks good to me, thanks! Is there a reason why the PR is still marked as a  `draft`?

**Comment by gregorydemay** [comment]
> Looks good to me, thanks! Is there a reason why the PR is still marked as a `draft`?

Sorry that was a mistake on my side, was ready to be reviewed but I forgot to unmark it as draft on Github

**Pull request merged**